core/src/gateway_internal.c

```diff
@@ -554,8 +554,11 @@ void gateway_removemodule_internal(GATEWAY_HANDLE_DATA* gateway_handle, MODULE_D
     remove_module_from_any_source(gateway_handle, *module_data_pptr);
     LINK_DATA *link;
     /* Codes_SRS_GATEWAY_26_018: [ This function shall remove any links that contain the removed module either as a source or sink. ] */
-    while ((link = VECTOR_find_if(gateway_handle->links, link_name_both_find, (*module_data_pptr)->module_name)) != NULL)
-        gateway_removelink_internal(gateway_handle, link);
+    if (gateway_handle->links)
```

Comment:
This is an internal function that is part of the code path for five public functions. Some enter the `if` clause and some don't. The result is effectively the same before and after this change, except now we don't print out misleading LogError statements. I don't see much value in a targeted unit test for this, beyond what we already have.